pkg/prometheus/promcfg.go

```diff
@@ -938,7 +938,9 @@ func (cg *ConfigGenerator) generateProbeConfig(
 		xc := labeler.GetRelabelingConfigs(m.TypeMeta, m.ObjectMeta, m.Spec.Targets.StaticConfig.RelabelConfigs)
 		relabelings = append(relabelings, generateRelabelConfig(xc)...)
 		cfg = append(cfg, yaml.MapItem{Key: "relabel_configs", Value: relabelings})
-	} else {
+	}
+
+	if m.Spec.Targets.Ingress != nil {
```

Comment:
This change was done because of this test case, this test case didn't exist before I added it, I'm happy to remove it and revert this change however, if a user creates a Probe without `Spec.Targets` this will lead to the operator crashing with a Seg Fault:

https://github.com/JoaoBraveCoding/prometheus-operator/blob/320ffa339d41c76c33c4416a939ba58c8238437e/pkg/prometheus/promcfg_test.go#L7853-L7868

Reply:
Nice catch! 

For a future PR, we could probably add `// +kubebuilder:validation:Required` markers. I'm not sure if blackbox-exporter works without targets, but, if it does, sounds like a mistake from the user

Reply:
IIUC validation markers can't check that at least one of `.spec.targets.static` or `.spec.targets.ingress`  is defined. It should be possible with [CEL](https://kubernetes.io/docs/reference/using-api/cel/) though but it's still [beta](https://kubernetes.io/blog/2022/09/23/crd-validation-rules-beta/). Anyway, supporting/using CEL is probably a discussion to have in a separate issue.

Reply:
In practice, the operator will not select a probe that have nil static and ingress fields.

https://github.com/prometheus-operator/prometheus-operator/blob/a487750b374b7c69fef1c8fee88af74c9cac8eb3/pkg/prometheus/resource_selector.go#L497C31-L500

https://github.com/prometheus-operator/prometheus-operator/blob/a487750b374b7c69fef1c8fee88af74c9cac8eb3/pkg/apis/monitoring/v1/probe_types.go#L113-L119

Reply:
I've added a comment to the beginning of the function